modules/json/tests/test_json_parser.c

```diff
@@ -151,61 +141,41 @@ test_json_parser_validate_type_representation(void)
   assert_log_message_value(msg, log_msg_get_value_handle(".prefix.array[1]"), "2");
   assert_log_message_value(msg, log_msg_get_value_handle(".prefix.array[2]"), "3");
   log_msg_unref(msg);
+  log_pipe_unref(&json_parser->super);
 }
 
-static void
-test_json_parser_fails_for_non_object_top_element(void)
+Test(json_parser, test_json_parser_fails_for_non_object_top_element, .init = setup, .fini = teardown)
 {
-  assert_json_parser_fails("[1, 2, 3]");
-  assert_json_parser_fails("");
+  LogParser *json_parser = json_parser_new(NULL);
+  assert_json_parser_fails("[1, 2, 3]", json_parser);
+  assert_json_parser_fails("", json_parser);
+  log_pipe_unref(&json_parser->super);
 }
 
-static void
-test_json_parser_extracts_subobjects_if_extract_prefix_is_specified(void)
+Test(json_parser, test_json_parser_extracts_subobjects_if_extract_prefix_is_specified, .init = setup, .fini = teardown)
 {
   LogMessage *msg;
+  LogParser *json_parser = json_parser_new(NULL);
 
   json_parser_set_extract_prefix(json_parser, "[0]");
-  msg = parse_json_into_log_message("[{'foo':'bar'}, {'bar':'foo'}]");
+  msg = parse_json_into_log_message("[{'foo':'bar'}, {'bar':'foo'}]", json_parser);
   assert_log_message_value(msg, log_msg_get_value_handle("foo"), "bar");
   log_msg_unref(msg);
+  log_pipe_unref(&json_parser->super);
 }
 
-static void
-test_json_parser_works_with_templates(void)
+Test(json_parser, test_json_parser_works_with_templates, .init = setup, .fini = teardown)
```

Comment:
You could define a `TestSuite`, in which case you do not have to repeat the `.init = setup, .fini = teardown` part.

Reply:
fixed